### PR TITLE
Add None as reduction option in CosineSimilarity

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/metrics/issues/398
Adds `None` as reduction option in cosine similarity to be consistent with codebase

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by FlorianMF** [comment]
nice, you did even update tests and added reduction value assertion.

**Comment by Borda** [comment]
@SkafteNicki still a draft?

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki mind check the failing tests?

File: torchmetrics/regression/cosine_similarity.py [hunk 1]
```diff
@@ -77,10 +77,13 @@ def __init__(
             process_group=process_group,
             dist_sync_fn=dist_sync_fn
         )
+        allowed_reduction = ('sum', 'mean', 'none', None)
+        if reduction not in allowed_reduction:
+            raise ValueError(f"Expected argument `reduction` to be one of {allowed_reduction}" f" but got {reduction}")
```
Comment (awaelchli) [inline torchmetrics/regression/cosine_similarity.py:82]:
```suggestion
            raise ValueError(f"Expected argument `reduction` to be one of {allowed_reduction} but got {reduction}")
```

**Review by awaelchli — APPROVED** [review]

**Pull request merged**